main/SAPI.c

```diff
@@ -1095,6 +1095,54 @@ SAPI_API void sapi_terminate_process(TSRMLS_D) {
 	}
 }
 
+SAPI_API void sapi_add_request_header(char *var, unsigned int var_len, char *val, unsigned int val_len, void *arg TSRMLS_DC) /* {{{ */
+{
+	zval *return_value = (zval*)arg;
+	char *str = NULL;
+	char *p;
+	ALLOCA_FLAG(use_heap)
+
+	if (var_len > 5 &&
+		var[0] == 'H' &&
+		var[1] == 'T' &&
+		var[2] == 'T' &&
+		var[3] == 'P' &&
+		var[4] == '_') {
```

Comment:
Is there anyway to actually preserve the header case? This naïve algorithm won't work properly for certain headers, and would break poorly-written code that relies on header case.